packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -2217,42 +2217,28 @@ class IterableProperty<T> extends DiagnosticsProperty<Iterable<T>> {
   );
 
   @override
-  String valueToString({ TextTreeConfiguration parentConfiguration }) {
+  String valueToString({TextTreeConfiguration parentConfiguration}) {
     if (value == null)
       return value.toString();
 
     if (value.isEmpty)
       return ifEmpty ?? '[]';
 
-    // Use debugFormatDouble to format Iterable<double>.
-    if (T == double) {
-      final StringBuffer sb = StringBuffer();
-      if (parentConfiguration != null && !parentConfiguration.lineBreakProperties) {
-        for (dynamic item in value) {
-          if (sb.isNotEmpty) {
-            sb.write(', ');
-          }
-          sb.write(debugFormatDouble(item));
-        }
-        return '[${sb.toString()}]';
+    final Iterable<String> formattedValues = value.map((T v) {
+      if (T == double && v is double) {
```

Comment:
@munificent It is unfortunate Flutter lints are requiring this check when all we want to do is check.
Can we get the T == double check to flow through or correct the Flutter lints to not worry about sealed types like double?

Reply:
Checking the type parameter for each element doesn't make much sense to me. It's a property of the surrounding class, so it's going to be the same each time. Why not just do `if (v is double)`?

That accomplishes everything you're trying to express. If you really want to check the type argument, you could do:

```
Iterable<String> formattedValues;
if (T == double) {
  formattedValues = value.map((T v) => debugFormatDouble(v as double));
} else {
  formattedValues = value.map((T v) => v.toString());
}
```

But you still have to do some kind of runtime type check per element because Dart doesn't have anything like template specialization.